packages/react-core/src/components/Skeleton/Skeleton.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Skeleton/skeleton';
+import { css } from '@patternfly/react-styles';
+
+export interface SkeletonProps extends React.HTMLProps<HTMLDivElement> {
+  /** Additional classes added to the Skeleton */
+  className?: string;
+  /** The percentage width of the Skeleton */
+  width?: string;
+  /** The percentage height of the Skeleton */
+  height?: string;
+  /** The font size height of the Skeleton */
+  fontSize?: 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl';
+  /** The shape of the Skeleton */
+  shape?: 'circle' | 'square' | 'rectangle';
+  /** Width of the shape */
+  shapeWidth?: 'sm' | 'md' | 'lg';
```

Comment:
I wonder if we need to separate out the shape width here.  From what I see in core, I would expect one prop for the width modifiers. 
I don't see anything that says that the ` 'sm' | 'md' | 'lg' ` only applies when a shape is set. Or, that you can not use percentage for when you use a shape.  @mcoker  is this the case?
For height I would expect a similar prop.
@mcarrano @mcoker do we want to allow custom heights and widths?

Reply:
@tlabaj yep and yep!

By default, the skeleton is the height of a line of text (the font-size * line-height), and will fill the width of the container it is in.

From there, a user can assign a height and/or a width. And those can be one of our preset modifiers, or they could pass something custom like `300px` or `42%` or whatever. That custom value can be used as the value for `--pf-c-skeleton--Width` and `--pf-c-skeleton--Height`. For example, `<Skeleton height="300px">...` renders `<div class="pf-c-skeleton" style="--pf-c-skeleton--Height: 300px">...`

And from there, a user can also assign a shape of either circle or square. When a shape is assigned, any custom height is discarded, and the height just matches the specified width. We could either make that part of the documentation, or we could do this if they have set a height with a circle/square shape:

* If a user has passed both a height and a width, we discard the height for them.
* If the user has specified either a custom height or one of our height modifiers, we make that the value of the width instead.

Reply:
@tlabaj @mcoker Updated with these changes:

- removed `shapeWidth`/`shapeHeight` => rolled into `width`/`height`
- updated custom width/height styling to use `--pf-c-skeleton--Width` and `--pf-c-skeleton--Height` (instead of style width and height)
- updated `width` and `height` to take any `%` or `px` or preset of `sm`|`md`|`lg`. But now it does need `%` or `px` in the string to work, and presets are only detected when using `%`.